tests/mocks/MockFileSystem.hh

```diff
@@ -53,6 +59,55 @@ public:
         static const std::string sEmpty;
         return sEmpty;
     }
+    
+    /// <summary>
+    /// Mocks the size of the file.
+    /// </summary>
+    /// <param name="sPath">The path of the file.</param>
+    /// <param name="nFileSize">The size of the file.</param>
+    /// <remarls>Mocked size will be discarded if the file is updated (MockFile, CreateTextFile, or AppendTextFile is called for the file).</remarks>
+    void MockFileSize(const std::wstring& sPath, int64_t nFileSize)
+    {
+        m_mFileSizes.insert_or_assign(sPath, nFileSize);
+    }
+
+    int64_t GetFileSize(const std::wstring& sPath) const override
+    {
+        auto pIter = m_mFileSizes.find(sPath);
+        if (pIter != m_mFileSizes.end())
+            return pIter->second;
+
+        auto pIter2 = m_mFileContents.find(sPath);
+        if (pIter2 != m_mFileContents.end())
+            return pIter2->second.length();
+
+        return -1;
+    }
+
+    bool DeleteFile(const std::wstring& sPath) const override
+    {
+        m_mFileSizes.erase(sPath);
+        return m_mFileContents.erase(sPath) != 0;
+    }
+
+    bool MoveFile(const std::wstring& sOldPath, const std::wstring& sNewPath) const override
+    {
+        auto hNode = m_mFileContents.extract(sOldPath);
+        if (hNode.empty())
+            return false;
+
+        hNode.key() = sNewPath;
+        m_mFileContents.insert(std::move(hNode));
+
+        auto hNode2 = m_mFileSizes.extract(sOldPath);
```

Comment:
This seems like it can destroyed early if the operation failed.

Reply:
Why are you so obsessed about destroying things? None of these methods are critical paths, and until a profiler proves otherwise, not worth optimizing. You might even argue that constantly shifting the stack around is detrimental to performance.

In this particular case, destroying it "early" will only exclude the return statement.